riscv/src/runtime.rs

```diff
@@ -148,6 +148,48 @@ impl Runtime {
         self.syscalls.contains_key(&s)
     }
 
+    pub fn with_keccak(mut self) -> Self {
+        self.add_submachine(
+            "test_data::asm::keccakf::KeccakF",
+            None,
+            "keccakf",
+            [format!(
+                "instr keccakf ~ keccakf.keccakf {};", // operation
+                instr_register_params(0, 25, 25) // 25 registers for the states, covered by 26 registers of riscv
+            )],
+            0,
+            // zero out output registers
+            // TODO: I did this similar to what poseidon did,
+            // but I'm concerned that any preexisting values in the registers aren't saved.
+            // Can I simply leave this blank?
```

Comment:
I think this is actually not needed because the machine will use memory directly.

Reply:
yes

Reply:
and the reason why we need 2 pointers, one for input and one for output, is that we can do both the input reads and the output writes in a single row

Reply:
Got it, and these two pointers are really both "input" argument pointers, i.e. the Rust wrapper will take two arguments, each corresponding to a pointer in the register? It's just that the function will operate on the array pointed by the pointer for the input, and put the results into the array pointed by the other pointer for the output?

Reply:
yes! As @pacheco also said in some other comment, the advantage of taking the output pointer also as parameter is that the user is then responsible for allocating the output array and our library doesn't have to do that

Reply:
Done.